src/mixins/canvas_events.mixin.js

```diff
@@ -819,12 +830,38 @@
      */
     _fireOverOutEvents: function(target, e) {
       this.fireSyntheticInOutEvents(target, e, {
-        targetName: '_hoveredTarget',
+        oldTarget: this._hoveredTarget,
         canvasEvtOut: 'mouse:out',
         evtOut: 'mouseout',
         canvasEvtIn: 'mouse:over',
         evtIn: 'mouseover',
       });
+      var _this = this;
+      this.targets.forEach(function(_target, index) {
```

Comment:
i was thinking to save space here:
```js
var _hoveredTarget = this._hoveredTarget, _hoveredTargets = this._hoveredTargets, targets = this.targets,
      diff = _hoveredTargets.length - targets.length;
[target].concat(targets, new Array(diff)).forEach(function(_target, index) {
        _this.fireSyntheticInOutEvents(_target, e, {
          oldTarget: _hoveredTargets[index - 1],  // because we know we shifted one place with the new array trick
          canvasEvtOut: 'mouse:out',
          evtOut: 'mouseout',
          canvasEvtIn: 'mouse:over',
          evtIn: 'mouseover',
        });
});
this._hoveredTarget = target;
this._hoveredTargets = this.targets.concat();
```

Reply:
oh we have to handle if diff is bigger than 0. But at least is more terse.
I often pull the variables away from `this` with their own name. I do that just because fabrics is big also when minimized and `this.hoveredTarget` does not get minimized. So if i use it more than once in a function, pulling it in a variable allow me to minimize it and save after the first occurrence.

Reply:
updated here: https://github.com/fabricjs/fabric.js/pull/6013/commits/426e68cb407bced5b8205be211339fcdb7de3631

Reply:
tweaked one aspect of this in my commit
`oldTarget: index === 0 ? _hoveredTarget : _hoveredTargets[index - 1],`